sentry/src/main/java/io/sentry/metrics/EncodedMetrics.java

```diff
@@ -0,0 +1,21 @@
+package io.sentry.metrics;
+
+import java.nio.charset.Charset;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+@ApiStatus.Internal
+public final class EncodedMetrics {
+  @SuppressWarnings({"CharsetObjectCanBeUsed"})
+  private static final Charset UTF8 = Charset.forName("UTF-8");
+
+  private final @NotNull String statsd;
+
+  public EncodedMetrics(@NotNull String statsd) {
```

Comment:
I would put the metrics serialization in this class, too, otherwise this is just a String, and the caller is responsible to properly instantiate it.